infrastructure/cli/pkg/providers/minikube/minikube.go

```diff
@@ -1,20 +1,124 @@
 package minikube
 
 import (
-	"cli/pkg/kube"
+	"bufio"
+	"cli/pkg/core"
+	"context"
 	"fmt"
-	"os"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/homedir"
+	"net/url"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	minikube = "minikube"
+	profile  = "airy-core"
 )
 
 type Minikube struct {
+	context       core.KubeCtx
+	hosts         core.AvailableHosts
+}
+
+func (m *Minikube) GetHelmOverrides() []string {
+	return []string{"--set", "global.enableNgrok=true"}
 }
 
-func (m *Minikube) Provision() (kube.KubeCtx, error) {
-	// Use this to
-	//clientcmd.NewNonInteractiveClientConfig()
-	fmt.Println("minikube provider not yet implemented")
-	os.Exit(1)
-	return kube.KubeCtx{}, nil
+func (m *Minikube) Provision() (core.KubeCtx, error) {
+	if err := checkInstallation(); err != nil {
+		return core.KubeCtx{}, err
+	}
+
+	if err := startCluster(); err != nil {
+		return core.KubeCtx{}, err
+	}
+
+	homeDir := homedir.HomeDir()
+	if homeDir == "" {
+		return core.KubeCtx{}, fmt.Errorf("could not find the kubeconfig")
+	}
+
+	ctx := core.New(filepath.Join(homeDir, ".kube", "config"), profile)
+	m.context = ctx
+	return ctx, nil
 }
 
+// Updates the host configmap with the service url
+func (m *Minikube) PostInstallation(namespace string) error {
+	clientset, err := m.context.GetClientSet()
+	if err != nil {
+		return err
+	}
+
+	configMaps := clientset.CoreV1().ConfigMaps(namespace)
+
+	configMap, err := configMaps.Get(context.TODO(), "hostnames", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	parsedUrl, err := url.Parse(m.hosts.Api.Url)
+	if err != nil {
+		return err
+	}
+
+	configMap.Data["HOST"] = parsedUrl.Host
+	_, err = configMaps.Update(context.TODO(), configMap, metav1.UpdateOptions{})
+
+	return err
+}
 
+func checkInstallation() error {
+	_, err := exec.LookPath(minikube)
+	return err
+}
+
+func startCluster() error {
+	return run("start", "--driver=virtualbox", "--cpus=4", "--memory=7168")
```

Comment:
I'd make this default arguments of provider specific flags (but can/should be done in a follow up)

Reply:
You are reffering to the `--provider-config=cpus=4` pattern that @ljupcovangelski suggested, right? I agree we can move them there once we have established that pattern

Reply:
yes (even though I think the flag names could be simpler... but no need to discuss here)